tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsContextPropagator.cs

```diff
@@ -31,7 +34,13 @@ public void Inject<TCarrier>(PathwayContext context, TCarrier headers)
     {
         if (headers is null) { ThrowHelper.ThrowArgumentNullException(nameof(headers)); }
 
-        headers.Add(DataStreamsPropagationHeaders.PropagationKey, PathwayContextEncoder.Encode(context));
+        var base64EncodedContext = Convert.ToBase64String(PathwayContextEncoder.Encode(context));
```

Comment:
You could call `PathwayContextEncoder.Encode(context)` only once and re-use that value for both headers instead of doing the work twice and allocating an additional `byte[]`.